### DISCUSSION TITLE
How to exclude logs for certain qnames

### DISCUSSION THREAD

**Discussion by rdsubhas** [coredns/coredns#7027] [Q&A]
We currently use the [log plugin](https://coredns.io/plugins/log/) with a lot of internal traffic. For internal traffic, this creates a huge amount of noise and stdout overhead.

It is unfortunately not possible to separate the internal traffic into a separate listener, because our upstream does search domain expansion, which means most of the requests end under fallback `.` and then get resolved via search domain expansion in the upstream DNS server.

Right now, the log plugin allows to cherry-pick what qnames to include. But this is impossible for us, because we want to log all the other lookups, which becomes practically infinite.

Is there any way to exclude or ignore certain qnames in the log plugin please?

**Comment by chrisohaver** [comment]
You can add a server block in the corefile for those names. The plugins in the server block would be the same as the original, except without the log plugin.

**Comment by rdsubhas** [comment]
Hi @chrisohaver it's explained in the comment above:

> It is unfortunately not possible to separate the internal traffic into a separate listener _(yes this is server block)_, because our upstream does search domain expansion, which means most of the requests go into the root `.` and then get resolved via search domain expansion in the upstream DNS server.
